### PR TITLE
Encapsulate running cargo into a struct

### PR DESCRIPTION
**Pull request by illicitonion**
I'm about to add conditionally setting environment variables depending on the cargo version in-use, so it's useful to route all calls to cargo through controlled methods.

### CODE REVIEW

File: crate_universe/src/metadata.rs [hunk 1]
```diff
@@ -80,6 +85,44 @@ impl MetadataGenerator for Generator {
     }
 }
 
+#[derive(Clone)]
+pub struct Cargo {
+    path: PathBuf,
+    full_version: Arc<Mutex<Option<String>>>,
```
Comment (UebelAndre) [inline crate_universe/src/metadata.rs:91]: Why not just `Option<String>`?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
A nit and a question but otherwise seems good 😄

File: crate_universe/src/metadata.rs (hunk 1 above)
Comment (illicitonion) [reply crate_universe/src/metadata.rs:91]: To do that, we'd need every method which takes a `Cargo` to take a mutable reference to it, which would require some futzing around. Instead, we can just pass around shared references, at the cost of acquiring a lock to check whether we've already looked up the full version.

**Review by UebelAndre — APPROVED** [review]
Looks good to me!

**Pull request merged**